templates/long.yaml

```diff
@@ -0,0 +1,18 @@
+# This expanded template is intentionally kept simple.
+# Each resource type is used as few times as possible.
+# Each use should be for testing different edge cases.
+# Edge cases should be described in the resource name.
+
+- name: Test Project - With Folders
+  type: Project
+  children:
+  - name: Test Folder - With Subfolder and One of Many
+    type: Folder
+    children:
+    - name: Test Subfolder - Empty
+      type: Folder
+  - name: Test Folder - Empty and One of Many
+    type: Folder
+
+- name: Test Project - Empty
```

Comment:
Even though we may have decided that multi project templates are out of scope, the array format allows us to do this.  I don't actually think it would be too much overhead in client code.  We are going to have to loop through the resources anyhow.

Reply:
What Tom pointed out is important.  Expanding on what Tom said, reading an array of objects is a task that's already performed in expanding the short version.  Rendering anchors and merges creates an array of documents anyway.  This means that it's likely that we won't need to do anything with the object that comes from the long YAML template before feeding it to the same function.

Reply:
@mattfazza: There's a lot to unpack here, so I want to break it down to make sure I understand what you're saying before merging this PR. 

Can you specify what in Tom's statement is important and why?

> Rendering anchors and merges creates an array of documents anyway. 
Did you mean "aliases" instead of "merges"? I'm not sure what you mean by "merges" otherwise. 

Also, could you clarify what you mean by "an array of documents"? Are you referring to "documents" in the [YAML spec](https://yaml.org/spec/1.2/spec.html#id2800132) sense (separated by `...` or `---`)?

Reply:
> Can you specify what in Tom's statement is important and why?

Tom said "`We are going to have to loop through top level resources anyhow`".  The last implementation I remember from the short template renders an array of objects, and we were looping through those already.   On line 17, making it a list, happens to be compatible to that.  

Array of documents == [{}, {}] .  I was referring to separate objects in the YAML.  I don't think we have written different documents in the same YAML for this yet.  Merges are not applicable here - at least we haven't written template with it.  I just tend to talk about anchors (&) along with merges.  My bad if that was confusing to you.

*
> Rendering anchors and merges creates an array of documents anyway.

I just tested it, and that statement is not necessarily true.  It's implementation dependent - which is good for us because it means that we can write a template that does/doesn't do that.

Reply:
@mattfazza: I was confused because I didn't know about merges in YAML. Could you point me to a page that describes them? A few things come up when I search for it, and I want to be sure we're talking about the same thing.